MSAL/src/native_auth/controllers/MSALNativeAuthBaseController.swift

```diff
@@ -115,6 +115,21 @@ class MSALNativeAuthBaseController {
         MSIDTelemetry.sharedInstance().flush(context.telemetryRequestId())
     }
 
+    func updateTelemetryEvent(
```

Comment:
Three comments about this method:
- The name doesn't clarify that the telemetry event would be always stopped. I'd include the word "stop" or "complete" to this method name
- Given that we're using a generic error message, this can make the debug a little bit more difficult
- Why we have 2 MSALNativeAuthError? The first one is in the "failure" result (line 121), while the second one is a default parameter (line 122). At line 126 why we use the variable error even if the result is successful?

Reply:
1. I agree that this is not a great name. I've renamed it as `stopTelemetryEvent` as an overload of the others that also stop telemetry, and changed the parameter names to make it clearer.
2. `error.errorDescription` contains the name of the method that hasn't been implemented. The log would be like this: `"Error MSALNativeAuth has called the delegate method onSignUpCompleted that has not been implemented"`
3. I've added documentation for this. In some cases the controller creates an error (for example the SignUpController in the `.attributeValidationFailed`. That error goes here. The `failure` result is related to the developer not having implemented the optional delegate.

```swift
func stopTelemetryEvent(
  _ event: MSIDTelemetryAPIEvent?,
  delegateDispatcherResult: Result<Void, MSALNativeAuthError>,
  context: MSIDRequestContext,
  controllerError: MSALNativeAuthError? = nil
)
```